utils_cv/detection/data.py

```diff
@@ -132,3 +140,339 @@ def coco_labels() -> List[str]:
+    * `annos[n]['Labeled Data']`: URL of the original image
+    * `annos[n]['View Label']`: URL of the image with labels or masks
+    * `annos[n]['Label']`: Dict.  Meta data of all annotations of the
+      image
+    * `annos[n]['Label']['objects']`: List.  Meta data of all objects of
+      the image.
+    * `annos[n]['Label']['objects'][0]['value']`: Object name (category)
+    * **`annos[n]['Label']['objects'][0]['instanceURI']`**: URL of the
+      binary mask of the object, with 0 as background, 255 as the object.
+
+    Take the
+    [`odFridgeObjects`](https://cvbp.blob.core.windows.net/public/datasets/object_detection/odFridgeObjects.zip)
+    dataset as an example.  Here the XML annotations are in the
+    `odFridgeObjects/annotations` folder and the original images are in
+    the `odFridgeObjects/images` folder.  For an arbitrary image
+    `odFridgeObjects/images/xyz.jpg`, its corresponding XML annotation
+    file is `odFridgeObjects/annotations/xyz.xml`.
+
+    Because the missing parts are the masks annotated in LabelBox, the
+    only thing we need to do is to combine all binary masks
+    (`[obj['instanceURI'] for obj in annos[0]['Label']['objects']]`) of
+    individual objects from an image (`annos[0]['External ID']`) into a
+    single mask image (`annos[0]['External ID'][:-4] + '.png'`) in a
+    directory called `segmentation-masks`.
+
+    Args:
+        labelbox_json: mask annotation JSON file from Labelbox
+        old_dir: path to dataset.  The path should contain the 'images' and
+            'annotations' subdirectories which store the original images and
+            PASCAL VOC annotation XML files.
+        new_dir: path to the result.  It will contain a 'segmentation-masks'
+            subdirectory as well as 'images' and 'annotations'.  Only images
+            with masks described in labelbox_json will be stored in new_dir.
+            Mask images extracted into 'segmentation-masks' will be PNG files.
+    """
+
+    old_img_dir = old_dir / 'images'  # image folder
+    old_anno_dir = old_dir / 'annotations'  # annotation folder
+
+    new_img_dir = new_dir / 'images'
+    new_anno_dir = new_dir / 'annotations'
+    new_mask_dir = new_dir / 'segmentation-masks'  # mask folder
+
+    # create directories for annotated dataset
+    new_img_dir.mkdir(parents=True, exist_ok=True)
+    new_anno_dir.mkdir(parents=True, exist_ok=True)
+    new_mask_dir.mkdir(parents=True, exist_ok=True)
+
+    # read exported LabelBox annotation JSON file
+    with open(labelbox_json) as f:
+        annos = json.load(f)
+
+    # process one image per iteration
+    for anno in annos:
+        # get related file paths
+        im_name = anno['External ID']  # image file name
+        anno_name = im_name[:-4] + '.xml'  # annotation file name
+        mask_name = im_name[:-4] + '.png'  # mask file name
+
+        print("Processing image: {}".format(im_name))
+
+        old_img = old_img_dir / im_name
```

Comment:
Variable names not clear / consistent with rest of CVBP. E.g. this is a path, not an image, hence old_im_path maybe, though I don't like "old" hence better "src_im_path" and "dst_im_path" maybe. This applies also to other places in this file.

Reply:
Renamed all similar variables throughout the code.